subgraphs/friend-tech/src/common/events.ts

```diff
@@ -0,0 +1,70 @@
+import { Address, Bytes, BigInt, ethereum } from "@graphprotocol/graph-ts";
+
+import { getOrCreateEthToken } from "./getters";
+import { BIGINT_TEN, ETH_DECIMALS, TradeType, ZERO_ADDRESS } from "./constants";
+import { NetworkConfigs } from "../../configurations/configure";
+
+import { Trade } from "../../generated/schema";
+
+export function createEvent(
+  traderAddress: Address,
+  subjectAddress: Address,
+  shares: BigInt,
+  sharePriceETH: BigInt,
+  subjectFeeETH: BigInt,
+  protocolFeeETH: BigInt,
+  tradeAmountETH: BigInt,
+  isBuy: boolean,
+  event: ethereum.Event
+): void {
+  const id = Bytes.empty()
+    .concat(event.transaction.hash)
+    .concatI32(event.logIndex.toI32());
+
+  const tradeEvent = new Trade(id);
+  tradeEvent.hash = event.transaction.hash;
+  tradeEvent.logIndex = event.logIndex.toI32();
+  tradeEvent.protocol = NetworkConfigs.getFactoryAddress();
+  tradeEvent.to = event.transaction.to
+    ? event.transaction.to!
+    : Address.fromString(ZERO_ADDRESS);
+  tradeEvent.from = event.transaction.from;
+
+  tradeEvent.trader = traderAddress;
+  tradeEvent.subject = subjectAddress;
+  tradeEvent.type = isBuy ? TradeType.BUY : TradeType.SELL;
+  tradeEvent.shares = shares;
+
+  const eth = getOrCreateEthToken(event);
+  const sharePriceUSD = sharePriceETH
```

Comment:
might be cleaner to have a helper functiuon to convert the native amount to the USD amount